src/app/border_agent.cpp

```diff
@@ -241,11 +241,11 @@ static int HandleRecord(const struct sockaddr *from,
                     ExitNow(error = ERROR_BAD_FORMAT("value of TXT Key 'sb' is invalid: value={}", utils::Hex(bitmap)));
                 }
 
-                borderAgent.mState.mConnectionMode = (bitmap[0] >> 5);
-                borderAgent.mState.mThreadIfStatus = (bitmap[0] << 3) >> 6;
-                borderAgent.mState.mAvailability   = (bitmap[0] << 5) >> 6;
-                borderAgent.mState.mBbrIsActive    = (bitmap[0] << 7) >> 7;
-                borderAgent.mState.mBbrIsPrimary   = (bitmap[1] >> 7);
+                borderAgent.mState.mConnectionMode = (bitmap[3] & 0x07);
```

Comment:
can we use a struct to better read each bits?

Reply:
If you are talking something about `borderAgent.mState = *reinterpret_cast<State*>(bitmap)`, then I would say this is not safe. Because the order of each bitfields is not defined by the C/C++ language, but implementation defined.

Reply:
Fine to keep it as is